src/editor/EditorManager.js

```diff
@@ -779,6 +790,22 @@ define(function (require, exports, module) {
         }
     }
 
+    function providerAvailableForPos(hostEditor) {
+        var providersFound = [],
+            pos = hostEditor.getCursorPos(),
+            providerRet,
+            i, len,
+            providers = _inlineEditProviders.concat(_inlineDocsProviders);
```

Comment:
I think we can simplify this, and make it less expensive by only looping through both if we need to, and then only returning `true/false`:

```js
for(i=0, len=_inlineEditProviders.length; i<len; i++) {
  if(_inlineEditProviders[i].queryProvider(hostEditor, pos)) {
    return true;
  }
}

for(i=0, len=_inlineDocsProviders.length; i<len; i++) {
  if(_inlineDocsProviders[i].queryProvider(hostEditor, pos)) {
    return true;
  }
}

return false;
```